### PR TITLE
Octave: clean up of patches, head does build, java disabled by default 

### PR DESCRIPTION
**Pull request by schoeps**
- patches are not inline but downloaded from savannah (easier to keep track)
- the development branch 4.x can be compiled
- Octave can be compiled with osmesa (but still segfaults)
- Octave can be build with audio, see #2875 
- disabled java by default, see #1899
- added comments how to print graphics via cairo

### CODE REVIEW

File: octave.rb [hunk 1]
```diff
@@ -22,108 +23,134 @@ class Octave < Formula
...
+
+    # arguments for options enabled by default
+    args << "--without-curl"            if build.without? "curl"
+    args << "--disable-docs"            if build.without? "docs"
+    args << "--without-fftw3"           if build.without? "fftw"
+    args << "--without-glpk"            if build.without? "glpk"
+    args << "--disable-gui"             if build.without? "gui"
+    args << "--without-hdf5"            if build.without? "hdf5"
+    args << "--without-opengl"          if build.without? "opengl"
+    args << "--without-framework-opengl"if build.without? "opengl"
```
Comment (dpo) [inline octave.rb]: There should really be a space before `if` ;-)

Comment (schoeps) [inline octave.rb]: spaces are overrated :)

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by schoeps** [comment]
I do not understand the issues of the TestBot. Any help appreciated.

**Comment by dpo** [comment]
The Brew test bot is happy. Travis just times out. If this is ready, please squash you commits.

**Comment by schoeps** [comment]
@dpo: is there any problem with this pull request?

**Comment by dpo** [comment]
@schoeps Sorry. Free time is the only problem.

**Comment by dpo** [comment]
@BrewTestBot test this please

**Pull request closed without merging**